### PR TITLE
Add example data to package and document the data generation and contents

### PR DESCRIPTION
**Pull request by taylorreiter**
Adds data, data provenance, and data documentation for example data that will be loaded with the package. I did my very best to follow all of the conventions laid out here: https://r-pkgs.org/data.html

Unfortunately, example data can't be more than 1M in size if we want to upload the package to CRAN one day (which I think is a good goal to have). Also, I wanted to make sure I chose something where I completely documented the generation of the data (all of the Arcadia-specific samples I'm working with right now, I haven't run gather on the raw reads but instead on assemblies, and I don't have the exact code and versions use to get the raw reads to their assemblies). For that reason, I chose the gut data set that I used here: https://taylorreiter.github.io/2022-07-28-From-raw-metagenome-reads-to-phyloseq-taxonomy-table-using-sourmash-gather-and-sourmash-taxonomy/. The only thing this won't show super well is the time series alluvial plot. For that, I'll add subsampled data from a time series later (I have about 100kb to work with until I reach my 1Mb limit for data).

I added both extdata so that I can show the `read*` functions and data objects so people can work with those directly and see what their formats look like without having to use a `read*` function.

In subsequent PRs, I'll use these data to write examples for each function and to write a vignette.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -33,3 +33,5 @@ Imports:
     tidyr,
     tidyselect,
     vegan
+Depends: 
+    R (>= 2.10)
```
Comment (mertcelebi) [inline DESCRIPTION:37]: Isn't this a super wide range? I think we're at R v 4 something now. Is that a problem?

File: R/data.R [hunk 2]
```diff
@@ -0,0 +1,142 @@
+#' Gut microbiome sourmash compare data frame
+#'
+#' A similarlity matrix between six stool microbiome shotgun metagenome samples.
```
Comment (mertcelebi) [inline R/data.R:3]:
```suggestion
#' A similarity matrix between six stool microbiome shotgun metagenome samples.
```

File: data-raw/01_gut_compare_df.R [hunk 3]
```diff
@@ -0,0 +1,2 @@
+gut_compare_df <- sourmashconsumr::read_compare_csv("inst/extdata/gut_compare.csv.gz")
```
Comment (mertcelebi) [inline data-raw/01_gut_compare_df.R:1]: How are these files created and/or downloaded? I don't think these are from the sourmash commands file right?

And what does `inst` stand for?

File: DESCRIPTION (hunk 1 above)
Comment (taylorreiter) [reply DESCRIPTION:37]: it IS a super wide range...these depends/imports are autogenerated by `devtools::use_package()`. I've been following the R package guide to a T so I think it's ok. Prior to my adding data, it didn't depend on any specific version of R, so I think something about the data format wasn't implemented until 2.10

File: data-raw/01_gut_compare_df.R (hunk 3 above)
Comment (taylorreiter) [reply data-raw/01_gut_compare_df.R:1]: > How are these files created and/or downloaded? I don't think these are from the sourmash commands file right?

Ya this PR touches a lot of files so it's kinda hard to follow, but I'll do my best at explaining it!
1. `inst/extdata` contains any non-R formatted data that might be useful for demonstrating how your package works. In this case, because the package reads in text files, I need those files to live in `inst/extdata`
1. The `data-raw` folder is used to document how R formatted data that is accessible to the package is generated (typically called by the user with the `data` function, so `data(gut_compare_df)`. In this case, the only thing I did to create those objects in R is read in a CSV vile to an R object, then write the R object out to an Rda file.
2. Given this, I wanted to also document where the csv files came from. I used the scripts `data-raw/00_sourmash_commands.sh` to do that. It documents the version of sourmash I used to generate the files and then documents the shell commands one would need to run to re-generate all of the files in `inst/extdata`.

> And what does `inst` stand for?

I don't actually know what `inst` stands for, but if data is in `inst/extadata` when the package is built, the external data is then available at `extdata` to the user. Following conventions laid out here: https://r-pkgs.org/data.html

File: data-raw/00_sourmash_commands.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+for infile in *csv
+do
+   gzip ${infile}
+done
+
+####################################################################################
+## RUN SOURMASH COMPARE
+####################################################################################
+
+sourmash compare -k 31 -o gut_compare.csv *sig
```
Comment (elizabethmcd) [inline data-raw/00_sourmash_commands.sh:94]: Does this file just stay in `data-raw` to document how the example data was generated or does it become part of the documentation/tutorial vignettes? I know vignettes are usually just focused on the R package docs but would be good to reference this somewhere in the official docs as well

File: data-raw/01_gut_taxonomy_annotate_df.R [hunk 5]
```diff
@@ -0,0 +1,2 @@
+gut_taxonomy_annotate_df <- sourmashconsumr::read_taxonomy_annotate(Sys.glob("inst/extdata/SRR*with-lineages*csv.gz"))
+usethis::use_data(gut_taxonomy_annotate_df, overwrite = TRUE, compress = "xz")
```
Comment (elizabethmcd) [inline data-raw/01_gut_taxonomy_annotate_df.R:2]: why are the `gut_gather_df`, `gut_signatures_df` and `gut_taxonomy_annotate_df` called in separate R script in the `data-raw` foler?

**Review by elizabethmcd — APPROVED** [review]
Only a couple of questions - I also think sometime in the near future the time-series example data should be added to show the utility of the alluvial plot. I think having both this gut example data and a separate time-series example would be good, even if the time-series one is pared down.

File: data-raw/00_sourmash_commands.sh (hunk 4 above)
Comment (taylorreiter) [reply data-raw/00_sourmash_commands.sh:94]: it just stays in `data-raw`. I've added a reference to it in the vignette! (the vignette itself is still WIP)

File: data-raw/01_gut_taxonomy_annotate_df.R (hunk 5 above)
Comment (taylorreiter) [reply data-raw/01_gut_taxonomy_annotate_df.R:2]: 🤷‍♀️ it was just the convention documented in the R package instruction to have one R script per data set that's created

Comment (elizabethmcd) [reply data-raw/01_gut_taxonomy_annotate_df.R:2]: interesting...
